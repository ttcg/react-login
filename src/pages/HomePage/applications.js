import React, { useEffect, useState } from 'react'
import { connect } from 'react-redux'
import isEmpty from 'is-empty'
import {
    Col,
    Container,
    FormCheck,
    FormGroup,
    Row
} from 'react-bootstrap';

import { LoadingSpinner } from '../../components/LoadingSpinner'
import {
    getUserApplications
} from '../../redux/user/user.action'
import ApplicationRow from './applicationRow'
import postToUrl from '../../utilities/postToUrl'

export function Applications({
    isLoading,
    accessToken,
    applications,
    getUserApplications
}) {

    const [postToOrigin, setPostToOrigin] = useState(true);

    useEffect(() => {
        getUserApplications();
    }, [getUserApplications]);

    const replaceUrlToPost = url => {
        const currentEnv = getEnvFromUrl(url);
        let stsUrlToReplace;

        switch (currentEnv) {
            case "staging":
                stsUrlToReplace = "https://staging.ttcg.com";
                break;
            case "preview":
                stsUrlToReplace = "https://se-preview.ttcg.com";
                break;
            default:
                stsUrlToReplace = "https://test.ttcg.com";
                break;
        }

        return url.replace(stsUrlToReplace, "http://localhost/")
    }

    const getEnvFromUrl = url => {
        for (let item of ['test', 'staging', 'preview']) {
            if (url.indexOf(item) > 0)
                return item;
        }
    }

    const handleAppButtonClick = (data) => {

        const params = {
            'UserId': data.applicationUserId,
            'ApplicationType': data.applicationTypeId,
            'identityToken': accessToken
        };

        const stsUrl = postToOrigin ? data.applicationUrl : replaceUrlToPost(data.applicationUrl) ;

        postToUrl(stsUrl, params);        
    }

    return (
        <>
            {isLoading
                ?
                <LoadingSpinner />
                :
                !isEmpty(applications) &&
                <Container>
                    <Row>
                        <Col md={8}>
                            {applications                                
                                .map((data, key) =>
                                    <ApplicationRow
                                        key={key}
                                        data={data}
                                        handleAppButtonClick={handleAppButtonClick} />
                                )}
                        </Col>
                        <Col md={4} style={{ paddingTop: '50px' }}>
                            <FormGroup controlId="formBasicChecbox">
                                <FormCheck type="checkbox" label="Post To Original Url?" checked={postToOrigin}
                                    onChange={e => setPostToOrigin(e.target.checked)} className="font-weight-bold" />
                                Uncheck if you want to post to localhost
                            </FormGroup>

                        </Col>

                    </Row>
                </Container>
            }
        </>
    );
}

const mapStateToProps = (state) => {
    return {
        isLoading: state.UserReducer.isLoading,
        applications: state.UserReducer.applications
    };
};

const mapDispatchToProps = {
    getUserApplications
}

export default connect(mapStateToProps, mapDispatchToProps)(Applications);