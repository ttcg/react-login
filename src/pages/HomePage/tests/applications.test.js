import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Applications } from '../applications';
import ApplicationRow  from '../applicationRow';

it('should render Applications', () => {

    const data = [
        {
            "applicationName": "Test App1",
            "applicationTypeId": 1,
        },
        {
            "applicationName": "Test App2",
            "applicationTypeId": 2,
        }
    ];

    const component = shallow(<Applications applications={data} />);

    expect(component.find(ApplicationRow).length).toEqual(2);
    expect(toJson(component)).toMatchSnapshot();
});