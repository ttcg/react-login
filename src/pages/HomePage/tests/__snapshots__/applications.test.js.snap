// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Applications 1`] = `
<Fragment>
  <Container
    fluid={false}
  >
    <Bootstrap(Row)>
      <Col
        md={8}
      >
        <ApplicationRow
          data={
            Object {
              "applicationName": "Test App1",
              "applicationTypeId": 1,
            }
          }
          handleAppButtonClick={[Function]}
          key="0"
        />
        <ApplicationRow
          data={
            Object {
              "applicationName": "Test App2",
              "applicationTypeId": 2,
            }
          }
          handleAppButtonClick={[Function]}
          key="1"
        />
      </Col>
      <Col
        md={4}
        style={
          Object {
            "paddingTop": "50px",
          }
        }
      >
        <FormGroup
          controlId="formBasicChecbox"
        >
          <FormCheck
            checked={true}
            className="font-weight-bold"
            disabled={false}
            inline={false}
            isInvalid={false}
            isValid={false}
            label="Post To Original Url?"
            onChange={[Function]}
            title=""
            type="checkbox"
          />
          Uncheck if you want to post to localhost
        </FormGroup>
      </Col>
    </Bootstrap(Row)>
  </Container>
</Fragment>
`;
